packages/blocks/src/api/registration.js

```diff
@@ -91,10 +96,16 @@ export function registerBlockType( name, settings ) {
 	}
 
 	settings = applyFilters( 'blocks.registerBlockType', settings, name );
+	if ( ! isPlainObject( settings ) ) {
+		console.error(
+			'Block settings need to remain a plain object after all "blocks.registerBlockType" filters are applied.'
```

Comment:
I'm not really following why this would specifically mention the filter when all subsequent console error logging just makes it a general statement about the shape of the `settings`. Is there a particular reason we suspect this might become corrupted as a result of the filter?

```suggestion
			'Block settings must be a valid object.'
```

Reply:
By design, we enforce the `object` structure in this API method. The only way to make it corrupted is through filters. I'm fine leaving this message more general, however, my intention was to make it more helpful.

Reply:
> By design, we enforce the `object` structure in this API method.

Ah, I see that now in the code above. 👍

Reply:
I picked your shorter message for simplicity :)